ucharclasses.sty

```diff
@@ -31,195 +30,230 @@
 %  The package options allow you to selectively enable certain unicode blocks
 % ----------------------------------------------------------------------------
 
-% We first define all blocks in a list together with their start and
-% end point
+% We first define all blocks in a list together with their start and end
+% point.
+%
+% Starting with XeTeX version 3.14159265-2.6-0.99994, the number of
+% \XeTeXcharclass registers was extended from 256 to 4096 entries; some not
+% so important blocks are thus provided only for this and newer versions.
+% The primitive \XeTeXinterwordspaceshaping was introduced by XeTeX 0.99994;
+% we use it as a flag to identify this version.
+%
+% However, LaTeX2e itself currently doesn't provide support for 4096
+% entries; we thus have to override the hard-coded limit.
+\ifdefined\XeTeXinterwordspaceshaping
```

Comment:
this is now probably a duplication of https://github.com/Pomax/ucharclasses/pull/12/files#diff-4cd68769b8d8e9f73de25f975e5e52fcL718